lib/vacancy_scraper/north_east_schools/scraper.rb

```diff
@@ -0,0 +1,176 @@
+require 'open-uri'
+require 'nokogiri'
+
+module VacancyScraper::NorthEastSchools
+  class Scraper
+    def initialize(url = 'https://www.jobsinschoolsnortheast.com/job/teacher-of-psychology-2/')
+      @vacancy_url = url
+    end
+
+    def page
+      @page ||= Nokogiri::HTML(open(@vacancy_url))
+      @page.search('br').each { |br| br.replace('\n') }
+      @page
+    end
+
+    # rubocop:disable Metrics/AbcSize
+    def map!
+      school = School.where('levenshtein(name, ?) <= 3 or url like ?', school_name, "#{url}%").first
+      return Rails.logger.debug("Unable to find school: #{school_name}") if school.nil?
+
+      vacancy = Vacancy.new
+      vacancy.job_title = job_title
+      vacancy.headline = job_title
+      vacancy.subject = Subject.find_by(name: subject)
+      vacancy.school = school
+
+      vacancy.job_description = Nokogiri::HTML(body.to_html).text
+      vacancy.working_pattern = working_pattern
+      vacancy.weekly_hours = work_hours
+      vacancy.minimum_salary = min_salary
+      vacancy.maximum_salary = max_salary
+      vacancy.pay_scale = PayScale.where(code: pay_scale).first
+      vacancy.starts_on = starts_on
+      vacancy.expires_on = ends_on
+      vacancy.status = min_salary.to_i.positive? ? :published : :draft
+      vacancy.publish_on = Time.zone.today
+
+      return vacancy.save if vacancy.valid?
+      Rails.logger.debug("Invalid vacancy: #{vacancy.errors.inspect}")
+    rescue StandardError => e
+      Rails.logger.debug("Unable to save scraped vacancy: #{e.inspect}")
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def vacancy
+      @vacancy ||= page.css('article.featured-vacancies').first
+    end
+
+    def job_title
+      vacancy.css('.page-title').text
+    end
+
+    def subject
+      subjects = Subject.pluck(:name).join('|')
+      job_title[/(#{subjects})/, 1]
+    end
+
+    def school_name
+      name = vacancy.xpath('//li[strong[contains(text(), "School:")]]').children.last.text.strip
+      name.tr("'", '%')
+    end
+
+    def contract
+      vacancy.xpath('//li[strong[contains(text(), "Contractual Status:")]]').children.last.text.strip
+    end
+
+    def working_pattern
+      pattern = vacancy.xpath('//li[strong[contains(text(), "Hours:")]]').children.last.text.strip
+      working_pattern = pattern[/(full.time|part.time)/i, 1]
+      working_pattern.downcase.starts_with?('f') ? :full_time : :part_time
+    rescue
+      :full_time
+    end
+
+    def work_hours
+      pattern = vacancy.xpath('//li[strong[contains(text(), "Hours:")]]').children.last.text.strip
+      pattern[/(\d+.\d+)/, 1]
+    rescue
+      nil
+    end
+
+    def salary
+      @salary ||= vacancy.xpath('//li[strong[contains(text(), "Salary:")]]').children.last.text.strip
+    end
+
+    def max_salary
+      max_salary = salary.scan(/\d*.?(\d\d+),(\d{3})/)
+      return max_salary[1].join('') if max_salary.present?
+
+      code = salary[/(UPS\d*)/, 1]
+      code = 'UPS3' if code == 'UPS'
```

Comment:
MPS - UPS appears to be used to it covers the entire school teacher payscale, so that would be the lowest on the main pay scale and the highest on the upper pay scale.

see here https://www.jobsinschoolsnortheast.com/job/lead-design-technology-required-september-2018-earlier-possible/ (even though its MPR and UPR (arg)